comp110-worksheet-6/OxoBoard.cs

```diff
@@ -6,42 +6,168 @@
-		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
+        public Mark[,] Squares;
+
+        // Constructor. Perform any necessary data initialisation here.
+        // Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
+        public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            //Sets up the boards values and sets them all to none
+            Squares = new Mark[3, 3];
+
+            for (int y = 0; y < 3; y++)
+            {
+                for (int x = 0; x < 3; x++)
+                {
+                    Squares[x, y] = Mark.None;
+                }
+            }
 		}
 
 		// Return the contents of the specified square.
 		public Mark GetSquare(int x, int y)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            Mark value = Squares[x, y];
+            return value;
 		}
 
 		// If the specified square is currently empty, fill it with mark and return true.
 		// If the square is not empty, leave it as-is and return False.
 		public bool SetSquare(int x, int y, Mark mark)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            bool taken;
+
+            if (Squares[x, y] == Mark.None)
+            {
+                Squares[x, y] = mark;
+                taken = true;
+            }
+
+            else
+            {
+                taken = false;
+            }
+
+            return taken;
 		}
 
 		// If there are still empty squares on the board, return false.
 		// If there are no empty squares, return true.
 		public bool IsBoardFull()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+            bool full;
+            int remaining = 0;
+
+            // Counts number of empty squares
+            for (int y = 0; y < 3; y++)
+            {
+                for (int x = 0; x < 3; x++)
+                {
+                    if (Squares[x, y] == Mark.None)
+                    {
+                        remaining++;
+                    }
+                }
+            }
+
+            // Returns true only if there are no empty squares
+            if (remaining > 0)
+            {
+                full = false;
+            }
+
+            else
+            {
+                full = true;
+            }
+
+            return full;
+        }
 
 		// If a player has three in a row, return Mark.O or Mark.X depending on which player.
 		// Otherwise, return Mark.None.
 		public Mark GetWinner()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            Mark winner = Mark.None;
+
+            // Checks if any X rows are all one type
```

Comment:
Careful of terminology here -- you talk about "X rows" and "Y rows" but also have variables `Xrow` and `Orow`, so it's not clear whether you mean X the coordinate or X the mark. Better to refer to rows and columns, or horizontal and vertical, or something less ambiguous like that